cdisc_rules_engine/models/dictionaries/abstract_dictionary_validator.py

```diff
@@ -0,0 +1,39 @@
+from abc import ABC, abstractmethod
+
+
+class AbstractDictionaryValidator(ABC):
+    @abstractmethod
+    def is_valid_term(
+        self, term_dictionary: dict, term: str, variable: str = "", **kwargs
+    ) -> bool:
+        """
+        Method to identify whether a term is valid based on its term type.
+
+        Args:
+            term_dictionary: The dictionary of available terms.
+            term: The dictionary term used
+            variable: The variable corresponding with the component of the
```

Comment:
to make this more generic, it seems like this parameter should correspond to the dictionary's variable (like PT), not the source data's variable (like DECOD).  Both the source variable (--DECOD) and the target dictionary variable (PT) should be specified with the operator in the rule logic.

For example,

```yaml
- operation: valid_external_dictionary_values
  name: --DECOD
  id: $is_valid_decod_value
  external_dictionary_type: meddra
  case_sensitive: false
  dictionary_variable: PT
```

Reply:
That makes sense, I will pass the source_variable and the target dictionary variable to the function though so we don't need any special mappings

Reply:
Updated. I added a dictionary_term_type param to the operation and pass both the source variable ("--DECOD") in this case and the proved dictionary_term_type to the is_valid_term method